packages/mui-material/src/Box/Box.d.ts

```diff
@@ -18,6 +18,6 @@ declare const Box: OverridableComponent<BoxTypeMap<{}, 'div', MaterialTheme>>;
 export type BoxProps<
   D extends React.ElementType = BoxTypeMap['defaultComponent'],
   P = {},
-> = OverrideProps<BoxTypeMap<P, D, MaterialTheme>, D>;
+> = OverrideProps<BoxTypeMap<P, D, MaterialTheme>, D> & { component?: D };
```

Comment:
@michaldudak after adding `{component?:D}` here - test from [Stack.spec.tsx](https://github.com/mui/material-ui/pull/38084/files#diff-85b8396d4b2c524e3d7f08b5b90a13e99d1395b64bbe9cb3450d253bd3c47994) file is failing. [here](https://app.circleci.com/pipelines/github/mui/material-ui/103023/workflows/dcc5b50b-ef9c-48c3-8c0b-f7f49cae1f1e/jobs/549789?invite=true#step-110-759) is the error from CI

Reply:
Box already has the `component` defined in its TypeMap (packages\mui-system\src\Box\Box.d.ts), so it's not necessary to add it here.

Reply:
okay fair but when i replaced `Box` with `Button` issue still persists. here is the codesandbox link : https://codesandbox.io/s/exciting-cloud-r4qvcm?file=/Demo.tsx. maybe underlying issue is different

Reply:
Ah, right. In this case, when we don't provide a type parameter to the ButtonProps, it defaults to `button`, and make the whole type too strict to be passed to the Stack's `component`.
Defining the `*Props` with `& { component?: React.ElementType }` instead would help here, but it would allow writing `<Button<'span'> component='div' />`. I think we can have such a compromise.

Reply:
I just committed this change, let's see how the CI will react.

Reply:
CI is green, shall I go ahead and add tests.

Reply:
Yes please!